src/internal/operators/share.ts

```diff
@@ -50,13 +87,65 @@ function shareSubjectFactory() {
  * // subscription 1:  9
  * // ... and so on
  * ```
- *
- * @see {@link api/index/function/interval}
- * @see {@link map}
- *
- * @return {Observable<T>} An Observable that upon connection causes the source Observable to emit items to its Observers.
- * @name share
  */
-export function share<T>(): MonoTypeOperatorFunction<T> {
-  return (source: Observable<T>) => refCount()(multicast(shareSubjectFactory)(source)) as Observable<T>;
+export function share<T, R>(options?: ShareOptions<T, R>): OperatorFunction<T, T | R> {
+  return (source: Observable<T>) => {
+    options = options || {};
+    let { connector, resetOnComplete = true, resetOnError = true, resetOnUnsubscribe = true } = options;
+    if (!connector) {
+      connector = () => new Subject<T>();
+    }
```

Comment:
isnt this more consistent?
```
const {
  connector = () => new Subject<T>(),
  resetOnComplete = true,
  resetOnError = true,
  resetOnUnsubscribe = true,
} = options || {};
```

also shouldnt this be moved into the `share` functions block insteadof the operator function block? maybe even:
```
export function share<T, R>({
  connector = () => new Subject<T>(),
  resetOnComplete = true,
  resetOnError = true,
  resetOnUnsubscribe = true,
}: ShareOptions<T, R> = {}): OperatorFunction<T, T | R> {
  // ...
}
```